### PR TITLE
Update docs on Error struct. #29355

### PR DESCRIPTION
**Pull request by rthomas**
This adds a pretty contrived example of the usage of fmt::Error. I am
very open to suggestions for a better one.

I have also highlighted the fmt::Error vs std::error::Error.

r? @steveklabnik

### CODE REVIEW

File: src/libcore/fmt/mod.rs
Comment (steveklabnik) [inline src/libcore/fmt/mod.rs]: I would say "An important thing to remember is" instead, that is, this is very important, but not *the* thing, IMO

File: src/libcore/fmt/mod.rs
Comment (steveklabnik) [inline src/libcore/fmt/mod.rs]: use std::fmt{self, write};

File: src/libcore/fmt/mod.rs
Comment (steveklabnik) [inline src/libcore/fmt/mod.rs]: match write

File: src/libcore/fmt/mod.rs
Comment (steveklabnik) [inline src/libcore/fmt/mod.rs]: instead of match, using `unwrap_or_else` is more idiomatic here

**Review by steveklabnik — COMMENTED** [review]
some nits. I agree that the example is not spectacular, but I have no better suggestions right now, and so it's fine.

File: src/libcore/fmt/mod.rs
Comment (QuietMisdreavus) [inline src/libcore/fmt/mod.rs]: If we're comparing to other `Error`s within the standard library, it might also be good to include `std::io::Error`, which I would consider more well-known than the Error trait.

File: src/libcore/fmt/mod.rs
Comment (rthomas) [inline src/libcore/fmt/mod.rs]: Done.

File: src/libcore/fmt/mod.rs
Comment (rthomas) [inline src/libcore/fmt/mod.rs]: Done.

File: src/libcore/fmt/mod.rs
Comment (rthomas) [inline src/libcore/fmt/mod.rs]: Done.

File: src/libcore/fmt/mod.rs
Comment (rthomas) [inline src/libcore/fmt/mod.rs]: I agree it would be more idiomatic, but that would not demonstrate the usage of `fmt::Error` - so I'm not sure if it would be worth including the example in that case.

File: src/libcore/fmt/mod.rs
Comment (rthomas) [inline src/libcore/fmt/mod.rs]: Done.

**Comment by rthomas** [comment]
I've pushed up a commit with the review comments addressed in it.

The only potentially open issue is the example.

**Comment by rthomas** [comment]
Friendly ping.

**Comment by steveklabnik** [comment]
Sorry about this! I took Monday of work to get a four-day weekend 😄 

I will review this later today.

File: src/libcore/fmt/mod.rs
Comment (steveklabnik) [inline src/libcore/fmt/mod.rs]: can you remove this please?

File: src/libcore/fmt/mod.rs
Comment (steveklabnik) [inline src/libcore/fmt/mod.rs]: This should be `Examples` even though there's only one, for consistency.

**Review by steveklabnik — CHANGES_REQUESTED** [review]
Hi! Sorry again. Two extremely tiny nits, and then let's :shipit: . Ping me once they're changed and I will merge post-haste!

**Comment by rthomas** [comment]
Thanks @steveklabnik I've fixed up the issues and pushed the commit up - would you like me to squash them down or are they fine as is?

**Comment by rthomas** [comment]
@steveklabnik friendly ping.

**Comment by steveklabnik** [comment]
@rthomas sorry, somehow I missed this!

Yeah, if you could squash, that'd be great. Happy to merge after 😄

**Review by steveklabnik — APPROVED** [review]

**Comment by rthomas** [comment]
Thanks @steveklabnik, I have squashed the commits.

**Comment by carols10cents** [comment]
@steveklabnik looks like this is ready for your r!

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thanks so much, and sorry again about the delays

**Pull request merged**
